frameworks/Python/starlette/app.py

```diff
@@ -0,0 +1,131 @@
+import asyncio
+import asyncpg
+import os
+import jinja2
+from starlette import (
+    asgi_application, HTMLResponse, JSONResponse, Path, PlainTextResponse, Router
+)
+from random import randint
+from operator import itemgetter
+from urllib.parse import parse_qs
+from ujson import dumps as ujson_dumps
+
+
+READ_ROW_SQL = 'SELECT "randomnumber" FROM "world" WHERE id = $1'
+WRITE_ROW_SQL = 'UPDATE "world" SET "randomnumber"=$1 WHERE id=$2'
+ADDITIONAL_ROW = [0, 'Additional fortune added at request time.']
+
+
+class UJSONResponse(JSONResponse):
+    def render(self, content):
+        return ujson_dumps(content).encode('utf-8')
+
+
+async def setup_database():
+    global connection_pool
+    connection_pool = await asyncpg.create_pool(
+        user=os.getenv('PGUSER', 'benchmarkdbuser'),
+        password=os.getenv('PGPASS', 'benchmarkdbpass'),
+        database='hello_world',
+        host='tfb-database',
+        port=5432
+    )
+
+
+def load_fortunes_template():
+    path = os.path.join('templates', 'fortune.html')
+    with open(path, 'r') as template_file:
+        template_text = template_file.read()
+        return jinja2.Template(template_text)
+
+
+def get_num_queries(request):
+    try:
+        query_string = request['query_string']
+        query_count = int(parse_qs(query_string)[b'queries'][0])
+    except (KeyError, IndexError, ValueError):
+        return 1
+
+    if query_count < 1:
+        return 1
+    if query_count > 500:
+        return 500
+    return query_count
+
+
+connection_pool = None
+sort_fortunes_key = itemgetter(1)
+template = load_fortunes_template()
+loop = asyncio.get_event_loop()
+loop.run_until_complete(setup_database())
+
+
+@asgi_application
+def json_serialization(request):
+    return UJSONResponse({'message': 'Hello, world!'})
+
+
+@asgi_application
+async def single_database_query(request):
+    row_id = randint(1, 10000)
```

Comment:
Not that it matters a whole lot, but the highest number this generates is 9999, correct?  If so, I think this should be `randint(1, 10001)`.  There is a record with id=10000 in the database.

Reply:
Nope, it’s 1-10000 inclusive.

Python’s docs for randint are: ‘Return a random integer N such that a <= N <= b’

(Unlike Pythons list slices)

Reply:
Oh my bad.  I was looking at the docs for numpy.random.randint.

Reply:
Thanks for the review @michaelhixson - Have a fab weekend! ✨